### PR TITLE
Add dict support on metadata, optimize datetime format judgment rules, add eq for combiner

### PR DESCRIPTION
**Pull request by MooooCat**
## Description

This PR mainly completes:
- Add dict support on metadata, for datetime inspector, the datetime_format returned by the update method can be passed in in list form;
- Optimize datetime format judgment rules, which increases robustness and reduces misjudgments in the face of missing data and incorrect data.

## How has this been tested?

The content of this modification has been covered by test cases.

## Types of changes

- [ ] Maintenance (no change in code, maintain the project's CI, docs, etc.)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: sdgx/data_models/inspectors/datetime.py [hunk 1]
```diff
@@ -85,19 +97,30 @@ def detect_datetime_format(self, series: pd.Series):
...
                str: The datetime format that can parse all dates in the series, or None if no such format is found.
         """
+
+        def _is_series_fit_format(parsed_series, match_rate):
+            length = len(parsed_series)
+            cnt = 0
+            for i in parsed_series:
+                if i is False:
+                    cnt += 1
+            false_rate = cnt / length
```
Comment (Wh1isper) [inline sdgx/data_models/inspectors/datetime.py:107]: We cloud make it simple

```suggestion
            false_num = len((for i in parsed_series if i is False))
            false_rate = false_mum / length
```

**Review by Wh1isper — APPROVED** [review]

File: sdgx/data_models/inspectors/datetime.py [hunk 2]
```diff
@@ -85,19 +97,27 @@ def detect_datetime_format(self, series: pd.Series):
         Returns:
                str: The datetime format that can parse all dates in the series, or None if no such format is found.
         """
+
+        def _is_series_fit_format(parsed_series, match_rate):
+            length = len(parsed_series)
+            false_num = len(list(i for i in parsed_series if i is False))
+            false_rate = false_mum / length
```
Comment (Wh1isper) [inline sdgx/data_models/inspectors/datetime.py:104]:
```suggestion
            false_rate = false_num / length
```

File: sdgx/data_models/inspectors/datetime.py [hunk 3]
```diff
@@ -85,19 +97,27 @@ def detect_datetime_format(self, series: pd.Series):
...
                str: The datetime format that can parse all dates in the series, or None if no such format is found.
         """
+
+        def _is_series_fit_format(parsed_series, match_rate):
+            length = len(parsed_series)
+            false_num = len(list(i for i in parsed_series if i is False))
+            false_rate = false_num / length
+            if false_rate >= match_rate:
+                return True
+            return False
```
Comment (Wh1isper) [inline sdgx/data_models/inspectors/datetime.py:107]:
```suggestion
            return false_rate >= match_rate
```

**Comment by MooooCat** [comment]
I had a breakfast before fix the code, I found that @Wh1isper  had already correct them when I push commits.  

hhhhhh, happy new year!

**Comment by Wh1isper** [comment]
Happy new year~

**Pull request merged**